core/embed/util/image/image.c

```diff
@@ -276,6 +275,11 @@ secbool check_image_contents(const image_header *const hdr, uint32_t firstskip,
     }
   }
 
+  // check hashes of image chunks
+  // we hash the image including the padding to the end of the area
+  offset = firstskip;
```

Comment:
...and this should be `size_t offset`, otherwise it's rather confusing why the same variable is redefined this way

Reply:
fixed https://github.com/trezor/trezor-firmware/commit/c4fe8423870981420e7da280af9b1d5f871fbfe8